src/models/Order.js

```diff
@@ -115,6 +117,30 @@ OrderSchema.pre('findOne', async function (next) {
   console.log('Pre-findOne finished');
   next();
 });
+
+// Middleware to send order completion email when the order status changes to "complete"
+OrderSchema.post('save', async function (doc) {
+  try {
+    if (doc.orderStatus === 'complete') {
```

Comment:
I didn't notice until now that we created both a `payment_successful` and a `complete` status. We probably only need one of those (or at least I can't picture currently what the difference between the two would be or in what case we would go from one to the next).

Reply:
An issue with this check is that `doc.orderStatus === "complete"` could be true even if the order status wasn't just updated in this query.

For example, imagine that we have an order that was already completed from weeks ago, but a request comes in to update the order total (maybe there was some bug or glitch that we're fixing now). The `post('save')` middleware would be called, and `doc.orderStats === 'complete'` would pass since the doucment's orderstatus does in fact match. Instead, we want to check if the query that just ran was to update the orderstatus specifically.

I'm not sure if you can do this with `post('save')` but  I know with `post('findOneAndUpdate')` we could check which parameters were passed to the update query with : `this.getUpdate().$set.orderStatus === 'payment_successful'`